File: commonpkg/models/item-model.go
package models

import (
	"time"

	"github.com/aws/aws-sdk-go/service/dynamodb"
)

type PurchaseDto struct {
	GodownId       string    `json:"godownId,omitempty"`
	SortKey        string    `json:"sortKey,omitempty"`        /// ProductId|QualityId|PurchaseId
	PurchaseId     string    `json:"purchaseId,omitempty"`     // GSI -1  PK  (all attr)
	PurchaseBillNo string    `json:"purchaseBillNo,omitempty"` // GSI - 2 PK (keys only)
	Date           time.Time `json:"date,omitempty"`
	VendorId       string    `json:"vendorId,omitempty"`
	ProductId      string    `json:"productId,omitempty"`
	QualityId      string    `json:"qualityId,omitempty"`
	Status         string    `json:"status,omitempty"` // Stock | Sold
}

type SalesDto struct {
	GodownId      string    `json:"godownId,omitempty"`
	SortKey       string    `json:"sortKey,omitempty"`     /// ProductId|QualityId
	SalesId       string    `json:"salesId,omitempty"`     // GSI -1  PK  (all attr)
	SalesBillNo   string    `json:"salesBillNo,omitempty"` // GSI PK (keys only)
	ClientId      string    `json:"clientId,omitempty"`
	TransporterId string    `json:"transporterId,omitempty"`
	LrNo          string    `json:"lrNo,omitempty"`
	ChallanNo     string    `json:"challanNo,omitempty"` // GSI PK (keys only)
	Date          time.Time `json:"date,omitempty"`
	ProductId     string    `json:"productId,omitempty"`
	QualityId     string    `json:"qualityId,omitempty"`
}

type BalePurchaseDetails struct {
	PurchaseId string
}
type BaleSalesDetails struct {
	SalesId string
}
type BaleTransferDetails struct {
	FromGodownId string    `json:"fromGodownId,omitempty"`
	ToGowodnId   string    `json:"toGowodnId,omitempty"`
	Date         time.Time `json:"date,omitempty"`
}

type BaleDetailsDto struct {
	GodownId         string                `json:"godownId,omitempty"`
	SortKey          string                `json:"sortKey,omitempty"` //// <InStock or OutOfStock>|ProductId|QualityId|BaleNo
	BaleNo           string                `json:"baleNo,omitempty"`  //GSI PK (all attr)
	ProductId        string                `json:"productId,omitempty"`
	QualityId        string                `json:"qualityId,omitempty"`
	BilledQuantity   int32                 `json:"billedQuantity,omitempty"`
	ReceivedQuantity int32                 `json:"receivedQuantity,omitempty"`
	Rate             int32                 `json:"rate,omitempty"`
	PurchaseDetails  BalePurchaseDetails   `json:"purchaseDetails,omitempty"`
	SalesDetails     BaleSalesDetails      `json:"salesDetails,omitempty"`
	TransferDetails  []BaleTransferDetails `json:"transferDetails,omitempty"`
}

type InventoryFilterDto struct {
	GodownId           string    `json:"godownId,omitempty"`
	PurchaseBillNumber string    `json:"purchaseBillNumber,omitempty" uri:"purchaseBillNumber"`
	SalesBillNumber    string    `json:"salesBillNumber,omitempty" uri:"salesBillNumber"`
	StartDate          time.Time `json:"startDate,omitempty"`
	EndDate            time.Time `json:"endDate,omitempty"`
	QualityId          string    `json:"qualityId,omitempty"`
	ProductId          string    `json:"productId,omitempty"`
}

type InventoryListRequest struct {
	LastEvalutionKey map[string]*dynamodb.AttributeValue `json:"lastEvalutionKey,omitempty"`
	PageSize         int64                               `json:"pageSize,omitempty" form:"pageSize"`
	PurchaseId       string
	PurchaseBillNo   string
	SalesId          string
	InventoryFilterDto
}

type PurchaseListResponse struct {
	CommonListResponse
	Data []PurchaseDto `json:"data,omitempty"`
}
type PurchaseResponse struct {
	CommonResponse
	Data PurchaseDto `json:"data,omitempty"`
}

type AddPurchaseDataRequest struct {
	PurchaseDetails PurchaseDto      `json:"purchaseDetails,omitempty"`
	BaleDetails     []BaleDetailsDto `json:"baleDetails,omitempty"`
}
type AddPurchaseDataResponse struct {
	CommonResponse
	PurchaseDetails PurchaseDto      `json:"purchaseDetails,omitempty"`
	BaleDetails     []BaleDetailsDto `json:"baleDetails,omitempty"`
}

// type InventoryResponse struct {
// 	CommonResponse
// 	Data InventoryDto `json:"data,omitempty"`
// }

// type BaleInfoReuest struct {
// 	GodownId string
// 	BaleNo   string `uri:"baleNo,omitempty"`
// 	Quality  string `uri:"quality,omitempty"`
// }

// type BaleInfoResponse struct {
// 	CommonResponse
// 	Purchase []BaleDetailsDto `json:"purchase,omitempty"`
// 	Sales    []BaleDetailsDto `json:"sales,omitempty"`
// }

File: ui/kp-textile/src/app/models/item-model.ts
import { CommonListResponse, CommonResponse } from "./genric-model"


export interface AddPurchaseDataRequest {
    purchaseDetails: PurchaseDto;
    baleDetails: BaleDetailsDto[];
}
export interface AddPurchaseDataResponse extends CommonResponse {
    purchaseDetails: PurchaseDto;
    baleDetails: BaleDetailsDto[];
}

export interface PurchaseDto {
    godownId: string;
    sortKey: string; /// ProductId|QualityId|PurchaseId
    purchaseId: string; // GSI -1  PK  (all attr)
    purchaseBillNo: string; // GSI - 2 PK (keys only)
    date: Date;
    vendorId: string;
    vendorName: string;
    productId: string;
    productName: string;
    qualityId: string;
    qualityName: string;
    hsnCode: string;
    purchaseStatus: string; // Stock | Sold
}

export interface SalesDto {
    godownId: string;
    sortKey: string;// ProductId|QualityId|SalesId
    salesId: string;// GSI -1  PK  (all attr)
    salesBillNo: string;// GSI PK (keys only)
    clientId: string;
    transporterId: string;
    lrNo: string;
    challanNo: string; // GSI PK (keys only)
    date: Date;
    productId: string;
    qualityId: string;
}

export interface BalePurchaseDetails {
    purchaseId: string;
}
export interface BaleSalesDetails {
    salesId: string;
}
export interface BaleTransferDetails {
    fromGodownId: string;
    toGowodnId: string;
    date: Date;
}

export interface BaleDetailsDto {
    godownId: string;
    sortKey: string;
    baleNo: string;
    productId: string;
    qualityId: string;
    billedQuantity: number;
    receivedQuantity: number;
    rate: number;
    purchaseDetails: BalePurchaseDetails;
    salesDetails: BaleSalesDetails;
    transferDetails: BaleTransferDetails[];
}

export interface PurchaseListResponse extends CommonListResponse{
	data:PurchaseDto[];
}

export interface BailDetailsDto {
    branchId: string;
    sortKey: string; //// Bail | <Purchase or Sales or OutOfStock>|Quality|BailNo|<salesBill or purchseBill number>
    bailNo: string;
    quality: string;
    isSales: boolean;
    billNo: string;
    rate: number;
    purchaseDate: Date;
    salesDate: Date;
    clientId: string;
    vendorId: string;
    transferedToBranchId: string;
    transferedFromBranchId: string;
    receivedQuantity: number;
    billedQuantity: number;
    pendingQuantity: number;
}

export interface BailInfoDto {
    branchId: string;
    bailInfoSortKey: string; /// Info | bailNo | Quality
    bailNo: string;
    receivedQuantity: number;
    billedQuantity: number;
    isLongation: boolean;
    quality: string;
}

export interface InventoryDto {
    branchId: string;
    inventorySortKey: string; /// Inventory | <Purchase or Sales>| Bill No
    billNo: string;
    bailDetails: BailDetailsDto[];
    purchaseDate: Date;
    salesDate: Date;
    transporterId: string;
    lrNo: string;
    challanNo: string;
    hsnCode: string;
}

export interface InventoryFilterDto {
    godownId: string
    purchaseBillNumber: string;
    salesBillNumber: string;
    qualityId: string;
    productId: string;
}

export interface InventoryListRequest extends InventoryFilterDto {
    lastEvalutionKey: any;
    pageSize: number;

}
export interface InventoryListResponse extends CommonListResponse {
    data: InventoryDto[];
}

export interface InventoryResponse extends CommonResponse {
    data: InventoryDto;
}

export interface BailInfoReuest {
    branchId: string;
    bailNo: string;
    quality: string;
}

export interface BailInfoResponse extends CommonResponse {
    purchase: BailDetailsDto[];
    sales: BailDetailsDto[];
}
